### ISSUE TITLE
FSSpecFileLister error with S3 root

### ISSUE THREAD

**Issue by nrlugg** [pytorch/data#348]
### 🐛 Describe the bug

The following code — listing files on S3 — fails:

```python
import torchdata.datapipes as dp

pipe = dp.iter.FSSpecFileLister(root="s3://my-bucket/...")
item = next(iter(pipe))
```
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/.../.venv/lib/python3.9/site-packages/torch/utils/data/_typing.py", line 366, in wrap_generator
    response = gen.send(None)
  File "/Users/.../.venv/lib/python3.9/site-packages/torchdata/datapipes/iter/load/fsspec.py", line 57, in __iter__
    is_local = fs.protocol == "file" or not self.root.startswith(fs.protocol)
TypeError: startswith first arg must be str or a tuple of str, not list
```

What's happening is that for the S3, `fs.protocol` is a `List[str]` (specifically `["s3", "s3a"]`) which is breaking `self.root.startswith(fs.protocol)` which excepts `Union[str, Tuple[str, ...]]`.
https://github.com/pytorch/data/blob/71a16a3b63966af0f60af74d3044b3c963cfc5fb/torchdata/datapipes/iter/load/fsspec.py#L55-L57

I *think* that the code expects `fs.protocol` to be a `str` all the time (i.e., not even a tuple), given the following line at the end if `FSSpecFileListerIterDataPipe.__iter__()`:
https://github.com/pytorch/data/blob/71a16a3b63966af0f60af74d3044b3c963cfc5fb/torchdata/datapipes/iter/load/fsspec.py#L74-L75
so a simple fix like replacing the list with a tuple might not work...

### Versions

```
Collecting environment information...
PyTorch version: 1.11.0
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 12.2.1 (arm64)
GCC version: Could not collect
Clang version: 13.1.6 (clang-1316.0.21.2)
CMake version: version 3.22.3
Libc version: N/A

Python version: 3.9.10 | packaged by conda-forge | (main, Feb  1 2022, 21:25:34)  [Clang 11.1.0 ] (64-bit runtime)
Python platform: macOS-12.2.1-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] mypy==0.941
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.22.3
[pip3] torch==1.11.0
[pip3] torch-cluster==1.6.0
[pip3] torch-geometric==2.0.4
[pip3] torch-scatter==2.0.9
[pip3] torch-sparse==0.6.13
[pip3] torch-spline-conv==1.2.1
[pip3] torchdata==0.3.0
[conda] mypy                      0.941                    pypi_0    pypi
[conda] mypy-extensions           0.4.3                    pypi_0    pypi
[conda] numpy                     1.22.3                   pypi_0    pypi
[conda] torch                     1.11.0                   pypi_0    pypi
[conda] torch-cluster             1.6.0                    pypi_0    pypi
[conda] torch-geometric           2.0.4                    pypi_0    pypi
[conda] torch-scatter             2.0.9                    pypi_0    pypi
[conda] torch-sparse              0.6.13                   pypi_0    pypi
[conda] torch-spline-conv         1.2.1                    pypi_0    pypi
[conda] torchdata                 0.3.0                    pypi_0    pypi
```

**Comment by nrlugg** [comment]
Since `fs.protocol` is a class variable, a *very* hacky workaround is:
```python
import s3fs

s3fs.core.S3FileSystem.protocol = "s3"
```
(assuming you know the filesystem will always be `"s3"`

**Comment by ejguan** [comment]
Thanks for finding such issue. Please feel free to open a PR to fix it.
And, we are currently trying to land a PR that uses `AWSSDK` to load `s3`. https://github.com/pytorch/data/pull/165 . But, in order to use it, you may have to compile `awssdk` by yourself.
We are still working on releasing a nightly wheel with awssdk library.

**Comment by nrlugg** [comment]
Will do! And thank for letting me know about `awssdk`

**Comment by ykabusalah** [comment]
@nrlugg 

I currently just submitted a solution to this issue as well in my solution to another issue. Could you please assign me to this issue?

**Comment by nrlugg** [comment]
@ykabusalah

Sorry for my slow reply... And thank you for solving this issue! It looks like you have already been assigned to this issue?

**Comment by NivekT** [comment]
Fixed by #383
